adapters/InMobi/InMobiAdapter/GADMAdapterInMobiUnifiedNativeAd.m

```diff
@@ -23,287 +30,301 @@ @interface GADMAdapterInMobiUnifiedNativeAd () <IMNativeDelegate>
+    
+    /// InMobi native ad object.
+    IMNative *_native;
+    
+    /// Indicates whether native ad images should be downloaded.
+    BOOL _shouldDownloadImages;
+    
+    /// Aspect ratio of the Native ad obtained from InMobi.
+    CGFloat _aspectRatio;
+    
+    /// Icon image sent to Google Mobile Ads SDK.
+    GADNativeAdImage *_mappedIcon;
+    
+    /// Array of GADNativeAdImage objects sent to Google Mobile Ads SDK.
+    NSArray<GADNativeAdImage *> *_mappedImages;
+    
+    /// A dictionary of asset names and object pairs for assets that are not handled by
+    /// properties of the GADMediatedUnifiedNativeAd.
+    NSDictionary<NSString *, id> *_extras;
+    
+    /// Contains the assests of the InMobi native ad.
+    NSDictionary<NSString *, id> *_nativeAdContentDictionary;
 }
 
 static NSCache *imageCache;
 
 __attribute__((constructor)) static void initialize_imageCache() {
-  imageCache = [[NSCache alloc] init];
+    imageCache = [[NSCache alloc] init];
 }
 
-- (nonnull instancetype)initWithGADMAdNetworkConnector:(nonnull id<GADMAdNetworkConnector>)connector
-                                               adapter:(nonnull id<GADMAdNetworkAdapter>)adapter {
-  self = [super init];
-  if (self) {
-    _connector = connector;
-    _adapter = adapter;
-    _shouldDownloadImages = YES;
-  }
-  return self;
+- (nonnull instancetype)init {
+    if (self = [super init]) {
+        _shouldDownloadImages = YES;
+    }
+    return self;
+}
+
+- (void)loadNativeAdForAdConfiguration:(nonnull GADMediationNativeAdConfiguration *)adConfiguration completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler {
+    
+    _nativeAdConfig = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationNativeLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _nativeRenderCompletionHandler = ^id<GADMediationNativeAdEventDelegate>(id<GADMediationNativeAd> nativeAd, NSError *error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationNativeAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(nativeAd, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    GADMAdapterInMobiUnifiedNativeAd *__weak weakSelf = self;
+    NSString *accountID = _nativeAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiUnifiedNativeAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
```

Comment:
`"InMobil SDK failed to initialize with error: %@"`

Reply:
done